### PR TITLE
Expose internals

### LINKED ISSUE #137: Expose internals
I'm working on an abstraction library for forking threads, and want to provide an `async` compatibility module for it.

Unfortunately, I cannot do this - the `Async` constructor is hidden, as is the `asyncUsing` function which I could use to construct it.

It would be Cool if the internals were avaiable in `Control.Concurrent.Async.Internal`.

### PR DESCRIPTION
**Pull request by parsonsmatt**
Fixes #137

### CODE REVIEW

**Comment by simonmar** [comment]
Hi @parsonsmatt, could you resolve the conflicts and update the PR please?

**Comment by simonmar** [comment]
resolved and pushed

File: Control/Concurrent/Async/Internal.hs [hunk 1]
```diff
@@ -0,0 +1,854 @@
+{-# LANGUAGE CPP, MagicHash, UnboxedTuples, RankNTypes,
+    ExistentialQuantification #-}
+#if __GLASGOW_HASKELL__ >= 701
+{-# LANGUAGE Trustworthy #-}
```
Comment (phadej) [inline Control/Concurrent/Async/Internal.hs:4]: IMO, `Internal` module cannot be `Trustworthy`. It's `Unsafe`.

**Comment by parsonsmatt** [comment]
Thank you! I apologize I wasn't able to get to this earlier.

**Pull request closed without merging**